core/src/main/java/com/datastax/oss/driver/api/core/tracker/RequestIdGenerator.java

```diff
@@ -67,6 +67,10 @@ default String getCustomPayloadKey() {
 
   default Statement<?> getDecoratedStatement(
       @NonNull Statement<?> statement, @NonNull String requestId) {
+    // in case of retry or speculative execution
+    if (statement.getCustomPayload().containsKey(getCustomPayloadKey())) {
```

Comment:
Should we not override the value instead of early-exiting? `traceparent` key will contain node request ID and IMHO it should be unique per attempt.